src/test/ui/proc-macro/span-api-tests.rs

```diff
@@ -25,7 +25,7 @@ reemit_legacy! {
     assert_source_file! { "Hello, world!" }
 }
 
-say_hello_extern! { assert_fake_source_file }
+say_hello_extern! { assert_source_file }
```

Comment:
Actually, now that I understand what this was doing (checking `proc_macro::SourceFile::is_real` returning `false`), maybe `say_hello_extern` should be replaced with a proc macro? cc @petrochenkov

Reply:
I think this may still be relevant.

Reply:
It's quite possible that proc macros are no longer "fake source files" either, since they keep their spans in metadata.
I agree that we need to attempt making `say_hello_extern` a proc macro, and if `assert_fake_source_file` still fails, then remove the test case.

Reply:
Answered in https://github.com/rust-lang/rust/pull/66364#issuecomment-590095860, spans from proc macros are always real.